### PR TITLE
fix(GraphQL) fix auth query rewriting with ID filter (cherry-pick-7740)

### PR DESCRIPTION
**Pull request by joshua-goldstein**
Refers to [this](https://github.com/dgraph-io/dgraph/pull/7740) PR.  Cherry pick 7740.

### CODE REVIEW

**Review by meghalims — APPROVED** [review]
Looks good 👍

**Comment by skrdgraph** [comment]
@joshua-goldstein - the tests seem to have failed, I am re-running them (because I suspect this must have happened when one of us was mucking with the runners)

**Comment by matthewmcneely** [comment]
Please don't merge this yet. Getting a panic when testing this locally.

**Comment by joshua-goldstein** [comment]
> Please don't merge this yet. Getting a panic when testing this locally.

Hi Matthew, any idea why this might be happening locally?  Do you have a stack trace of the panic?  I reran the test on our runners three times and we see the test passing there.

File: graphql/resolve/query_rewriter.go [hunk 1]
```diff
@@ -1567,7 +1573,18 @@ func idFilter(filter map[string]interface{}, idField schema.FieldDefinition) []u
 	if idsFilter == nil {
 		return nil
 	}
-	idsSlice := idsFilter.([]interface{})
```
Comment (matthewmcneely) [inline graphql/resolve/query_rewriter.go:1570]: This was panic-ing if the type coming in from the wild was a simple string—which I'd say is the typical scenario for IDs when encoded in JWTs (taken from some session ID or something).

**Comment by matthewmcneely** [comment]
@joshua-goldstein, @skrdgraph Could you guys review my recent commits to this branch?

**Comment by matthewmcneely** [comment]
> > Please don't merge this yet. Getting a panic when testing this locally.
> 
> Hi Matthew, any idea why this might be happening locally? Do you have a stack trace of the panic? I reran the test on our runners three times and we see the test passing there.

@joshua-goldstein The issue was that the tests were not creating all the conditions in which an ID-assigned JWT variable was being passed—namely, the tests covered arrays (slices), but not single values. I uncovered this by actually deploying a graph with auth protected types and running queries with actual JWTs in the requests.

So the fix was to check for that single value, not just the slice. I added another test to cover this condition.

**Review by skrdgraph — APPROVED** [review]
thnx for explaining this, it was easy to understand this @matthewmcneely

**Review by nswamy — APPROVED** [review]

**Pull request merged**